xcbgen-rs/src/defs.rs

```diff
@@ -1115,11 +1140,33 @@ impl Alignment {
 // FIXME: Make the members non-pub and make the constructor enforce the power-of-2 requirement.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub struct VariableSize {
-    pub base: u32,
-    pub incr: u32,
+    base: u32,
+    incr: u32,
 }
 
 impl VariableSize {
+    /// Creates a new `VariableSize` with `base` and `incr`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `incr` is not zero or a power of 2.
+    pub fn new(base: u32, incr: u32) -> Self {
+        assert!(incr == 0 || incr.is_power_of_two());
```

Comment:
Huh? If `incr` is always zero, why is it a parameter at all?

This sounds like it creates a "fixed size" instance of `VariableSize`, so what do you think about `for_fixed_size(base)` instead of `new(base, incr)` (perhaps you have a better idea for the name).

Reply:
Sorry, bad wording, changed to "neither zero nor a power of 2"

Reply:
Oh, my bad. I actually read `&&` instead of `||`.